File: SDKs/amazonaws.com/swf/2012-01-25/java/lib/src/main/java/openapisdk/models/shared/CancelTimerFailedEventAttributes.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * CancelTimerFailedEventAttributes
 * Provides the details of the <code>CancelTimerFailed</code> event.
**/
public class CancelTimerFailedEventAttributes {
    @JsonProperty("cause")
    public CancelTimerFailedCauseEnum cause;
    public CancelTimerFailedEventAttributes withCause(CancelTimerFailedCauseEnum cause) {
        this.cause = cause;
        return this;
    }
    @JsonProperty("decisionTaskCompletedEventId")
    public Long decisionTaskCompletedEventId;
    public CancelTimerFailedEventAttributes withDecisionTaskCompletedEventId(Long decisionTaskCompletedEventId) {
        this.decisionTaskCompletedEventId = decisionTaskCompletedEventId;
        return this;
    }
    @JsonProperty("timerId")
    public String timerId;
    public CancelTimerFailedEventAttributes withTimerId(String timerId) {
        this.timerId = timerId;
        return this;
    }
}

File: SDKs/amazonaws.com/swf/2012-01-25/python/src/sdk/models/shared/canceltimerfailedeventattributes.py
import dataclasses
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import canceltimerfailedcause_enum as shared_canceltimerfailedcause_enum


@dataclass_json
@dataclasses.dataclass
class CancelTimerFailedEventAttributes:
    r"""CancelTimerFailedEventAttributes
    Provides the details of the <code>CancelTimerFailed</code> event.
    """
    
    cause: shared_canceltimerfailedcause_enum.CancelTimerFailedCauseEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('cause') }})
    decision_task_completed_event_id: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('decisionTaskCompletedEventId') }})
    timer_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('timerId') }})
    
